src/notifications/notificationpreviewpresenter.cpp

```diff
@@ -172,18 +193,40 @@ bool NotificationPreviewPresenter::notificationShouldBeShown(LipstickNotificatio
         mode = surface->windowProperties().value("NOTIFICATION_PREVIEWS_DISABLED", uint(AllNotificationsEnabled)).toUInt();
     }
 
-    return !notificationHidden && notificationHasPreviewText && (!screenOrDeviceLocked || notificationIsCritical) &&
-            (mode == AllNotificationsEnabled || (mode == ApplicationNotificationsDisabled && notificationIsCritical) || (mode == SystemNotificationsDisabled && !notificationIsCritical));
+    return (!screenOrDeviceLocked || notificationIsCritical) &&
+            (mode == AllNotificationsEnabled ||
+             (mode == ApplicationNotificationsDisabled && notificationIsCritical) ||
+             (mode == SystemNotificationsDisabled && !notificationIsCritical));
 }
 
 void NotificationPreviewPresenter::setCurrentNotification(LipstickNotification *notification)
 {
     if (currentNotification != notification) {
         if (currentNotification) {
+            const bool displayWasOn(currentNotification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool());
+            if (displayWasOn) {
+                // Release our screen wake
+                QDBusMessage msg = QDBusMessage::createMethodCall(MCE_SERVICE, MCE_REQUEST_PATH, MCE_REQUEST_IF, MCE_NOTIFICATION_END);
+                msg.setArguments(QVariantList() << QString::number(currentNotification->replacesId()) << MCE_LINGER_DURATION);
+                QDBusConnection::systemBus().asyncCall(msg);
```

Comment:
No, I don't believe so.  They time out by themselves, or you can explicitly remove them prior to timeout.